action/index.d.ts

```diff
@@ -0,0 +1,24 @@
+import { WritableAtom, MapStore } from '../index.js'
+
+/**
+ * It's wraping your function and mark it as action for plugins.
+ * ```js
+ * const setName = action(store, 'setName', name => store.set(name))
+ * setName('John')
+ * ```
+ * @param store Store instance.
+ * @param actionName Loging name.
+ * @param cb Your own function.
+ * @returns Your own function.
+ */
+export function action<
+  Store extends WritableAtom,
+  ActionName extends string,
```

Comment:
We can use `store: WritableAtom, actionName: string` below.

We need to create this type templates for the cases where two arguments share the single type like:

```ts
<Objects extends object, Key extends keyof Object>(store: Object, key: Key)
```